internal/env/env.go

```diff
@@ -0,0 +1,90 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2025 Datadog, Inc.
+
+package env
+
+import (
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/DataDog/dd-trace-go/v2/internal/log"
+)
+
+// Getenv is a wrapper around env.Getenv that validates the environment variable
+// against a list of supported environment variables.
+//
+// If the environment variable has aliases, the function will also check the aliases
+// and return the value of the first alias that is set.
+//
+// When a environment variable is not supported because it is not
+// listed in the list of supported environment variables, the function will log an error
+// and behave as if the environment variable was not set.
+//
+// In testing mode, the reader will automatically add the environment variable
+// to the configuration file.
+func Getenv(name string) string {
+	if !verifySupportedConfiguration(name) {
+		return ""
```

Comment:
I can see that this will be a problem for the enhanced config telemetry work we will do. 

We need to report telemetry on all configurations that are set, not just the valid ones. In the future, we might need to change this API & LookupEnv to return the raw value and an error in this case.

Just a heads up.